src/nvim/main.c

```diff
@@ -1494,9 +1498,16 @@ static void init_params(mparm_T *paramp, int argc, char **argv)
 /// Initialize global startuptime file if "--startuptime" passed as an argument.
 static void init_startuptime(mparm_T *paramp)
 {
+  bool is_embedded_process = false;
+  for (int i = 1; i < paramp->argc - 1; i++) {
+    if (STRICMP(paramp->argv[i], "--embed") == 0) {
+      is_embedded_process = true;
```

Comment:
`init_startuptime` is called after `init_params`. Can you use the existing `embedded_mode` instead of re-scanning the args?

Reply:
maybe I'm confused but `init_params` just initializes the `paramp` struct to default values and doesn't actually parse anything.

```
static void init_params(mparm_T *paramp, int argc, char **argv)
{
  CLEAR_POINTER(paramp);
  paramp->argc = argc;
  paramp->argv = argv;
  paramp->use_debug_break_level = -1;
  paramp->window_count = -1;
  paramp->listen_addr = NULL;
  paramp->server_addr = NULL;
  paramp->remote = 0;
  paramp->luaf = NULL;
  paramp->lua_arg0 = -1;
}
```
https://github.com/pabloariasal/neovim/blob/fix_startup_time/src/nvim/main.c#L1484

Reply:
Ah sorry I was thinking of `command_line_scan`. 

I guess we can't move this down, since the profiling needs to start here. We could set `embedded_mode` here and remove the duplicate logic from `command_line_scan` ... not sure if that helps anything. 

Can leave this as-is for now.